### DISCUSSION TITLE
Conversion to C# / .NET ?

### DISCUSSION THREAD

**Discussion by codewhisperer86** [USEPA/EPANET2.2#68] [Announcements]
Is there much interest in converting this project to a managed language like C#?  This would make some of the other enhancements (such as UI enhancements) much more straightforward, as well as integrations with other technologies.  I'm willing to assist with this effort.

**Comment by cbuahin** [comment]
@codewhisperer84, good question. Unless things have changed recently, C# cannot be compiled to run natively on a machine. Therefore, rewriting the engine to C# will come with some performance hits. You could write C# bindings to the engine if you would like. I don't know if it is reasonable to recreate legacy codebases such as this one just so it can be closely bound to the UI. There are several C/C++ GUI frameworks (e.g., Qt) that would accomplish the same goal without going through the non-trivial task of rewriting the engine that would allow integration of other technologies. We are exploring these areas.

**Comment by cbuahin** [comment]
This is not an issue per se and will fit the better as a discussion item. I will move it there.
